src/main/java/gameframework/motion/MoveStrategyKeyboard.java

```diff
@@ -74,22 +77,23 @@ public void keyReleased(int keyCode) {
 	}
 
 	public void goRight() {
-		speedVector.setDirection(new Point(1, 0));
+		speedVector = new SpeedVector(new Point(1, 0));
```

Comment:
I don't understand what you are doing. With the code you propose, the speed is always reset to the default speed instead of keeping the current speed. Why do you want that? If you have a real use case, please explain it in the pull request and avoid changing current behavior.